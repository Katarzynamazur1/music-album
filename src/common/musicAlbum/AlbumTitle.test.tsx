import { render, RenderResult } from "@testing-library/react";
import AlbumTitle from "./AlbumTitle";

describe("AlbumTitle.tsx", () => {
  it("render result maches snapshot", () => {
    // Chcemy sprawdzic czy component <AlbumTitle /> sie wogole wyswietla
    //    Pytanie 1: Jak uzywamy AlbumTitle    Odp: <AlbumTitle>xxx</AlbumTitle>
    //    Pytanie 2: Czego mam oczekiwac po wyswietleniu <AlbumTitle>xxx</AlbumTitle> ?
    //      a) xxx
    //      b) <p>xxx</p>
    //      c) <div></div>
    //      d) <div>xxx</div>  (chyba prawidlowa odpowiedz :)

    // Problem: Musimy dac testowi mozliwosc wyswietlenia komponentu, niezaleznie od srodowiska, w ktorym jest odpalony.

    // Rozwiazanie problemu : rtl  = react testing library  ( z tej bilblioteki bierzemy redenrera, ktory pozwoli testowi wykorzystac implementacje, ktora dziala tak jak przegladarka www)
    const renderResult: RenderResult = render(<AlbumTitle>xxx</AlbumTitle>);
    expect(renderResult.container).toMatchSnapshot(); // to zapisalo nam plik ze snapshotem w katalogu __snapshots__
  });
});
