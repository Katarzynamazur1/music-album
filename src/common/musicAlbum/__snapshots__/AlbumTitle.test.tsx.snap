// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumTitle.tsx render result maches snapshot 1`] = `
<div>
  <div
    style="color: rgba(216, 237, 92, 0.612); font-weight: bold; font-size: 18px; font-family: 'Courier New', monospace; width: 330px; text-overflow: ellipsis; overflow: hidden; white-space: nowrap;"
  >
    xxx
  </div>
</div>
`;
